toponetx/transform/graph_to_simplicial_complex.py

```diff
@@ -93,3 +95,45 @@ def graph_2_clique_complex(
         stacklevel=2,
     )
     return graph_to_clique_complex(G, max_dim)
+
+
+def weighted_graph_to_vietoris_rips_complex(G, r, max_dim=None):
+    """Get the Vietoris-Rips complex of radius r of a weighted undirected graph.
+
+    The Vietoris-Rips complex of radius
+    r is the clique complex given by the cliques of G whose nodes have pairwise distances less or equal than r. All
+    vertices are added to the Vietoris-Rips complex regardless the radius introduced. If G is a clique weighted by a
+    dissimilarity function d that satisfies max_v d(v, v) <= min d(u,v) for u != v, and r >= d(v, v) for all nodes v,
+    then the Vietoris-Rips complex of radius r is the usual Vietoris-Rips abstract simplicial complex of radius r for
+    point clouds with dissimilarities.
+
+    Parameters
+    ----------
+    G : networkx graph
+        Weighted undirected input graph. The weights of the edges must be in the attribute 'weight'.
+    r : float
+        The radius for the Vietoris-Rips simplicial complex computation.
+    max_dim : int, optional
+        The max dimension of the cliques in
+        the output clique complex.
+        The default is None indicate max dimension.
+
+    Returns
+    -------
+    SimplicialComplex
+        The Vietoris-Rips simplicial complex of dimension max_dim of the graph G.
+    """
+
+    def is_in_vr_complex(clique):
+        edges = combinations(clique, 2)
+        edge_weights_lower_than_r = all(G[u][v]["weight"] <= r for u, v in edges)
+        return edge_weights_lower_than_r
+
+    all_cliques = nx.enumerate_all_cliques(G)
+    possible_cliques = (
+        all_cliques
+        if max_dim is None
+        else filter(lambda face: len(face) <= max_dim, all_cliques)
```

Comment:
Please don't use `filter` here, it is inefficient for low `max_dim`. See `graph_to_clique_complex` in this file and #221.

Reply:
Done